src/IRCSock.cpp

```diff
@@ -148,6 +150,70 @@ void CIRCSock::ReadLine(const CString& sData) {
 
 	DEBUG("(" << m_pNetwork->GetUser()->GetUserName() << "/" << m_pNetwork->GetName() << ") IRC -> ZNC [" << sLine << "]");
 
+	std::unique_ptr<int, std::function<void(int*)>> clearTagsHelper;
+
+	if (sLine.StartsWith("@")) {
+		m_sCurrentTagLine = sLine.Token(0);
+		m_sCurrentTagLine.LeftChomp();
+
+		sLine = sLine.Token(1, true);
+
+		m_msCurrentTags.clear();
+
+		std::vector<CString> vsTags;
+		m_sCurrentTagLine.Split(";", vsTags);
+
+		for (const CString &sTag: vsTags) {
+			CString sKey = sTag.Token(0, false, "=");
+			CString sRawValue = sTag.Token(1, true, "=");
+			CString sValue;
+
+			sValue.reserve(sRawValue.size());
+
+			char pc = 0;
+			for (const char &c: sRawValue) {
+				if (pc == '\\' && c == '\\') {
+					sValue += '\\';
+				} else if (pc == '\\' && c == ':') {
+					sValue += ';';
+				} else if (pc == '\\' && c == 's') {
+					sValue += ' ';
+				} else if (pc == '\\' && c == 'n') {
+					sValue += '\n';
+				} else if (pc == '\\' && c == 'r') {
+					sValue += '\r';
+				} else if (pc == '\\') {
+					sValue += pc;
+					sValue += c;
+				} else if (c != '\\') {
+					sValue += c;
+				}
+
+				if (pc == '\\' && c == '\\') {
+					pc = 0;
+				} else {
+					pc = c;
+				}
+			}
+
+			if (pc == '\\') {
+				sValue += pc;
+			}
+
+			m_msCurrentTags[sKey] = sValue;
+		}
+
+		clearTagsHelper = std::unique_ptr<int, std::function<void(int*)>>(new int, [this](int *v) {
+			delete v;
```

Comment:
Yes, that's what second type parameter for std::unique_ptr does. But you don't need int for this.